### PR TITLE
Extract import logic in class

### PR DESCRIPTION
**Pull request by FabianDevel**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/addFiles/AddFileBottomSheetDialog.kt [hunk 1]
```diff
@@ -110,12 +109,19 @@ class AddFileBottomSheetDialog : BottomSheetDialogFragment() {
...
+
+        uploadFileHelper = UploadFileHelper(
+            fragment = this@AddFileBottomSheetDialog,
+            parentFolder = currentFolderFile,
+            onOpeningPicker = {
+                trackNewElement("uploadFile")
+                binding.documentUpload.isEnabled = false
+            },
+            onResult = { findNavController().popBackStack() },
+        )
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/addFiles/AddFileBottomSheetDialog.kt:121]: Is it possible to initialize uploadFileHelper using a `by lazy` ?

File: app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        onOpeningPicker: () -> Unit,
+        onResult: (() -> Unit)? = null,
+    ) : this(parentFolder, fragment.findNavController(), onOpeningPicker, onResult) {
+        filePicker = FilePicker(fragment)
+        uploadFilesPermissions = DrivePermissions().apply {
+            registerPermissions(fragment) { authorized -> if (authorized) uploadFiles() }
+        }
+    }
+
+    constructor(
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt:54]: It seems that constructor is not used. Maybe it'll be used in another PR ?

File: app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+class UploadFileHelper private constructor(
+    private val parentFolder: File,
+    private val navController: NavController,
+    private val onOpeningPicker: () -> Unit,
+    private val onResult: (() -> Unit)?,
+) {
+
+    private lateinit var filePicker: FilePicker
+    private lateinit var uploadFilesPermissions: DrivePermissions
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt:40]: `uploadFilesPermissions` or `uploadFilePermissions` ? Sice the class is called `UploadFileHelper`.

File: app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import com.infomaniak.lib.core.utils.FilePicker
+
+class UploadFileHelper private constructor(
+    private val parentFolder: File,
+    private val navController: NavController,
+    private val onOpeningPicker: () -> Unit,
+    private val onResult: (() -> Unit)?,
+) {
+
+    private lateinit var filePicker: FilePicker
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt:39]: You can put this in the constructor to avoid having a lateinit.

**Review by tevincent — CHANGES_REQUESTED** [review]
Questions

File: app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt (hunk 2 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt:54]: Yes this is used in the next PR, but I really wanted to extract the logic all at once to have a full vision

File: app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt (hunk 3 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/ui/addFiles/UploadFileHelper.kt:40]: I renamed the class to UploadFileHelper

File: app/src/main/java/com/infomaniak/drive/ui/addFiles/AddFileBottomSheetDialog.kt (hunk 1 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/ui/addFiles/AddFileBottomSheetDialog.kt:121]: Only if I dissociate the `parentFolder` argument of the constructor in an init function, but then we would need to avoid forgetting calling this init, thus I prefered to keep it in constructor

**Review by tevincent — APPROVED** [review]
